aiven/aiven-operator
# add support for kafka topic

To test this please do the following:

- I recommend running K8s on your local machine as [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/), any other k8s distribution should do https://kubernetes.io/docs/tasks/tools/. 
- First, run `make install` and then `make run ENABLE_WEBHOOKS=false`
- Create a secret with Aiven API Token for your K8s cluster `kubectl create secret generic aiven-token --from-literal='token=${AIVEN_TOKEN}'`
- To test Kafka Topic we would need to create a Kafka service first, you can do that by running: 
```shell script
kubectl apply -f <(echo " 
apiVersion: k8s-operator.aiven.io/v1alpha1
kind: Kafka
metadata:
  name: my-k8s-kafka1

spec:
  service_name: my-k8s-kafka1
  project: dev-sandbox 
  cloud_name: google-europe-west1
  plan: business-4
  maintenance_window_dow: friday
  maintenance_window_time: 23:00:00
  kafka_user_config: {
    kafka_version: '2.4'
  }
")
```
- When Kafka service is reconciled aka up-and-running add a Kafka topic by running:
```shell script
kubectl apply -f <(echo " 
apiVersion: k8s-operator.aiven.io/v1alpha1
kind: KafkaTopic
metadata:
  name: my-k8s-topic1

spec:
  service_name: my-k8s-kafka1
  project: dev-sandbox 
  topic_name: my-k8s-topic1
  partitions: 3
  replication: 2
  tags: [
  	{
  		key: tagk1,
  		value: tagv1
  	},
  	{
  		key: tagk2,
  		value: tagv2
  	}
  ]
  config: {
  	flush_ms: 100
  }
")
```
